opennms-services/src/test/java/org/opennms/netmgt/poller/PollerRoutingTest.java

```diff
@@ -0,0 +1,139 @@
+    {
+        return new RouteBuilder() {
+
+            @Override
+            public void configure() throws Exception
+            {
+                // Add exception handlers
+                onException( IOException.class ).handled( true ).logStackTrace( true ).stop();
+
+                from( "direct:pollAvailabilityMonitor" ).to( "bean:availabilityMonitorCamel" ).split( body() ).recipientList(
+                                simple( "seda:Location-${body.location}.Poller.AvailabilityMonitor" ) );
+
+                from( "seda:Location-localhost.Poller.AvailabilityMonitor" ).to( "bean:availabilityMonitor" );
+            }
+        };
+    }
+    
+
+    @Test
+    public void testPoller() throws Exception
+    {
+        for ( RouteDefinition route : new ArrayList<RouteDefinition>( context.getRouteDefinitions() ) )
+        {
+            route.adviceWith( context, new AdviceWithRouteBuilder() {
+                @Override
+                public void configure() throws Exception
+                {
+                    mockEndpoints();
+                }
+            } );
+        }
+        context.start();
+
+        MockNetwork m_network = new MockNetwork();
+        m_network.setCriticalService("ICMP");
+        m_network.addNode(1, "Router");
+        m_network.addInterface("192.168.1.1");
+        m_network.addService("ICMP");
+        m_network.addService("SMTP");
+        m_network.addService("SNMP");
+        m_network.addInterface("192.168.1.2");
+        m_network.addService("ICMP");
+        m_network.addService("SMTP");
+        m_network.addNode(2, "Server");
+        m_network.addInterface("192.168.1.3");
+        m_network.addService("ICMP");
+        m_network.addService("HTTP");
+        m_network.addService("SMTP");
+        m_network.addService("SNMP");
+        m_network.addNode(3, "Firewall");
+        m_network.addInterface("192.168.1.4");
+        m_network.addService("SMTP");
+        m_network.addService("HTTP");
+        m_network.addInterface("192.168.1.5");
+        m_network.addService("SMTP");
+        m_network.addService("HTTP");
+        m_network.addNode(4, "DownNode");
+        m_network.addInterface("192.168.1.6");
+        m_network.addService("SNMP");
+        m_network.addNode(5, "Loner");
+        m_network.addInterface("192.168.1.7");
+        m_network.addService("ICMP");
+        m_network.addService("SNMP");
+    
+        
+        MockDatabase m_db = new MockDatabase();
+        m_db.populate(m_network);
+        DataSourceFactory.setInstance(m_db);
+        
+        PollerConfiguration m_pollerConfig = new PollerConfiguration();
+        
+        m_pollerConfig.setNextOutageId(m_db.getNextOutageIdStatement());
+        m_pollerConfig.setNodeOutage(new NodeOutage());
+        m_pollerConfig.addPackage(new Package());
+     
+
+        // Execute the job
+        template.requestBody( "direct:pollAvailabilityMonitor", m_pollerConfig );
```

Comment:
The pollAvailabilityMonitor endpoint should handle MonitoredService messages instead of PollerConfiguration messages. That is a difference between the poller tasks and the discovery tasks.

Also, the message should have the polling parameters, a map of key-value pairs that are sent to the monitor when the poll is executed. This is the second argument to the poll() method:
```
public PollStatus poll(MonitoredService svc, Map<String, Object> parameters);
```
You need to figure out how to pass the parameters into the poll() method. To do this, you may need to create a new message type that contains the MonitoredService and the parameters in one object and then add a new poll method, for instance:
```
public PollStatus poll(MonitoredServiceTask task) {
   return poll(task.getMonitoredService(), task.getParameters());
}
```